src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -168,14 +173,30 @@ void ConfigurationManagerImpl::RunConfigUnitTest(void)
 void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
 {
     ChipLogProgress(DeviceLayer, "Performing factory reset");
+
+#ifdef CONFIG_CHIP_FACTORY_RESET_ERASE_NVS
```

Comment:
Shouldn't we do it always, instead of creating two ways of doing factory reset? I mean, if this solution is more robust than current one, maybe we should just replace it.

Reply:
This approach is more robust but less flexible - if a manufacturer wanted to store some factory data or other device-lifelong settings it would erase them, too. Therefore, we should probably keep both options.